ApplicationCode/ModelVisualization/RivElementVectorResultPartMgr.cpp

```diff
@@ -271,51 +409,90 @@ void RivElementVectorResultPartMgr::createResultColorTextureCoords(
 
-    cvf::Vec3f headBottom = headTop - ( headTop - shaftStart ) * 0.2f;
+    // A fixed size is preferred here
+    cvf::Vec3f headBottom = headTop - ( headTop - shaftStart ).getNormalized() * headLength;
+    // cvf::Vec3f headBottom = headTop - ( headTop - shaftStart ) * 0.2f;
 
-    cvf::Vec3f headBottomDirection = evrViz.faceNormal ^ evrViz.faceCenter;
-    cvf::Vec3f arrowBottomSegment  = headBottomDirection.getNormalized() * headWidth;
+    cvf::Vec3f headBottomDirection1 = evrViz.faceNormal ^ evrViz.faceCenter;
+    cvf::Vec3f headBottomDirection2 = headBottomDirection1 ^ evrViz.faceNormal;
+    cvf::Vec3f arrowBottomSegment1  = headBottomDirection1.getNormalized() * headLength / 8.0f;
+    cvf::Vec3f arrowBottomSegment2  = headBottomDirection2.getNormalized() * headLength / 8.0f;
 
     vertices[0] = shaftStart;
     vertices[1] = headBottom;
-    vertices[2] = headBottom + arrowBottomSegment;
-    vertices[3] = headBottom - arrowBottomSegment;
+    vertices[2] = headBottom + arrowBottomSegment1;
+    vertices[3] = headBottom - arrowBottomSegment1;
     vertices[4] = headTop;
+    vertices[5] = headBottom + arrowBottomSegment2;
+    vertices[6] = headBottom - arrowBottomSegment2;
 
     return vertices;
 }
 
 //--------------------------------------------------------------------------------------------------
 ///
 //--------------------------------------------------------------------------------------------------
-std::array<uint, 8> RivElementVectorResultPartMgr::createArrowIndices( uint startIndex ) const
+std::array<uint, 2> RivElementVectorResultPartMgr::createArrowShaftIndices( uint startIndex ) const
 {
-    std::array<uint, 8> indices;
+    std::array<uint, 2> indices;
 
     indices[0] = startIndex;
     indices[1] = startIndex + 1;
-    indices[2] = startIndex + 2;
-    indices[3] = startIndex + 3;
-    indices[4] = startIndex + 3;
-    indices[5] = startIndex + 4;
-    indices[6] = startIndex + 4;
-    indices[7] = startIndex + 2;
 
     return indices;
 }
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::array<uint, 6> RivElementVectorResultPartMgr::createArrowHeadIndices( uint startIndex ) const
+{
+    std::array<uint, 6> indices;
+
+    indices[0] = startIndex + 2;
+    indices[1] = startIndex + 3;
+    indices[2] = startIndex + 4;
+
+    indices[3] = startIndex + 5;
+    indices[4] = startIndex + 6;
+    indices[5] = startIndex + 4;
+    return indices;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+double RivElementVectorResultPartMgr::scaleLogarithmically( double value ) const
+{
+    if ( value <= 1.0 )
+    {
+        value += 1.0;
```

Comment:
Not sure I understand this.

Reply:
If values are smaller than one, the logarithm would return increasing negative values the smaller the number is. However, small numbers shall remain small and not be scaled up. In order to achieve this, add 1.0 to small values in order to still obtain small positive numbers after scaling. I add a respective comment.